packages/client/components/promptResponse/PromptResponseEditor.tsx

```diff
@@ -80,22 +108,48 @@ const PromptResponseEditor = (props: Props) => {
     handleSubmit?.(newEditorState)
   }
 
+  const onCancel = (editor: EditorState) => {
+    setEditing(false)
+    editor?.commands.setContent(content)
+  }
+
   const editor = useEditor(
     {
       content,
       extensions: createEditorExtensions(placeholder),
       autofocus: autoFocus,
       onUpdate,
-      onBlur: onSubmit,
       editable: !readOnly
     },
     [content]
   )
 
   return (
-    <StyledEditor>
-      <EditorContent editor={editor} />
-    </StyledEditor>
+    <>
+      <StyledEditor>
+        <EditorContent editor={editor} />
+      </StyledEditor>
+      {!readOnly && (
+        // The render conditions for these buttons *should* only be true when 'readOnly' is false, but let's be explicit
+        // about it.
+        <SubmissionButtonWrapper>
+          {!!content && isEditing && (
```

Comment:
+1 `hasContent` would make this easier to read

Reply:
@BartoszJarocki could you elaborate a bit on the desired change here?  `content` here is the actual editor content, not a boolean.

Reply:
I actually have a weak preference towards leaving this as `!!content` since the use of `!!` is common enough for converting truthy/falsy things to true/false